lib/statsd.js

```diff
@@ -154,6 +154,45 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
     global.statsd = this;
   }
 
+  // only for uds (options.protocol uds)
+  // enabled with the extra flag options.udsGracefulErrorHandling
+  // will gracefully (attempt) to re-open the socket with a small delay
+  // options.udsGracefulRestartRateLimit is the minimum time (ms) between creating sockets
+  // does not support options.isChild (how to re-create a socket you didn't create?)
+  if (!options.isChild && options.protocol === 'uds' && options.udsGracefulErrorHandling) {
+    const socketCreateLimit = options.udsGracefulRestartRateLimit || 1000; // only recreate once per second
+    const lastSocketCreateTime = Date.now();
+    this.socket.on('error', (err) => {
+      const code = err.code;
+      switch (code) {
+        case 107:
+        case 111: {
+          if (Date.now() - lastSocketCreateTime >= socketCreateLimit) {
+            // recreate the socket, but only once per 30 seconds
+            if (this.errorHandler) {
+              this.socket.removeListener('error', this.errorHandler);
+            }
+            this.socket.close();
+            this.socket = createSocket(this, {
+              host: this.host,
+              path: options.path,
+              port: this.port,
+              protocol: this.protocol
+            });
```

Comment:
`createSocket(instance, args)` w. `args.protocol = 'uds'` only looks at `args.path`. Why then pass `host` and `path`?